homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -6,58 +6,105 @@
 import voluptuous as vol
 
 from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
     CONF_ADDRESS,
     CONF_DEVICE,
     CONF_NAME,
     DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
     POWER_WATT,
+    TEMP_CELSIUS,
 )
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.entity import Entity
+
+from .aurora_device import AuroraDevice
+from .const import ATTR_SERIAL_NUMBER, DEFAULT_ADDRESS, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_ADDRESS = 2
-DEFAULT_NAME = "Solar PV"
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_DEVICE): cv.string,
         vol.Optional(CONF_ADDRESS, default=DEFAULT_ADDRESS): cv.positive_int,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_NAME, default="Solar PV"): cv.string,
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Aurora ABB PowerOne device."""
+def setup_platform(hass, config: ConfigEntry, add_entities, discovery_info=None):
+    """Set up single sensor based on configuration.yaml (DEPRECATED)."""
     devices = []
     comport = config[CONF_DEVICE]
-    address = config[CONF_ADDRESS]
-    name = config[CONF_NAME]
-
+    address = config.get(CONF_ADDRESS, DEFAULT_ADDRESS)
+    name = config.get(CONF_NAME, "Solar PV")
+    _LOGGER.warning(
+        "DEPRECATED: setting up %s via configuration.yaml will "
+        "soon be unsupported.  Please remove the entries from your "
+        "configuration.yaml file and the set up the integration via the UI",
+        DOMAIN,
+    )
     _LOGGER.debug("Intitialising com port=%s address=%s", comport, address)
     client = AuroraSerialClient(address, comport, parity="N", timeout=1)
 
-    devices.append(AuroraABBSolarPVMonitorSensor(client, name, "Power"))
+    config.data = {"device_name": name}
+    config.entry_id = "undefined_entry_id"
+    devices.append(AuroraSensor(client, config, "Power", "instantaneouspower"))
     add_entities(devices, True)
 
 
-class AuroraABBSolarPVMonitorSensor(Entity):
-    """Representation of a Sensor."""
+async def async_setup_entry(hass, config, async_add_entities) -> None:
+    """Set up aurora_abb_powerone sensor based on a config entry."""
+    entities = []
+    client = hass.data[DOMAIN][config.entry_id]["client"]
+    serialnum = config.data[ATTR_SERIAL_NUMBER]
+    for sensor in hass.data[DOMAIN][config.entry_id]["devices"][serialnum]["sensor"]:
```

Comment:
why are you reading this data that you stored in it in `__init__.py` ? Define it here and then iterate over it

Reply:
Great idea! Thank you - now implemented this way.